docs/RFCS/20190906_flow_control.md

```diff
@@ -0,0 +1,131 @@
+# Proposal: Add flow control for data replication to downstream
+
+- Author(s):    [yangfei](https://github.com/amyangfei)
+- Last updated: 2019-09-18
+
+## Abstract
+
+Too fast import data or replication in DM may lead to downstream TiDB too busy to serve as normal. This proposal aims to find a way to detect the congestion of import or replication data flow and introduces a framework of dynamic concurrency control, which will help to control the import and replication data flow.
+
+Table of contents:
+
+- [Background](#Background)
+- [Implementation](#Implementation)
+    - [Congestion definition](#Congestion-definition)
+        - [How to measure the key indices](#How-to-measure-the-key-indices)
+        - [How to detect and measure congestion](#How-to-detect-and-measure-congestion)
+    - [Congestion control framework](#Congestion-control-framework)
+        - [The congestion detect component](#The-congestion-detect-component)
+        - [The data flow control component](#The-data-flow-control-component)
+        - [The concurrency control component](#The-concurrency-control-component)
+
+## Background
+
+Firstly we will discuss the key factor that affects the import or replication speed between DM data processing unit and downstream TiDB.
+
+- In the full data import procedure, data import speed is based on the SQL batch size and SQL executor concurrency. The SQL batch size is mainly determined by the single statement size of dump data, which can be controlled with the `mydumper` configuration. The SQL executor concurrency equals to the worker count of load unit, which is configurated by loader `pool-size` in task config.
+- In the incremental data replication procedure, replication speed relates to SQL job batch size and SQL executor concurrency. The binlog replication unit gets data from relay log consumer (assuming the relay log consume speed is fast enough), and distributes SQL jobs to multiple executors, the executor count is configurated by `worker-count` in task config. When the cached SQL jobs' count exceed SQL batch size, all SQLs in these jobs will be executed to downstream in a transaction.
```

Comment:
> When the cached SQL jobs' count exceed SQL batch size

talk about `batch` config item?

and we also have a `waitTime` (10ms) to execute SQL jobs when SQL batch size is not exceeded, talk about it?